apps/src/code-studio/pd/application/facilitator1819/Section3LeadingStudents.jsx

```diff
@@ -86,4 +86,25 @@ export default class Section3LeadingStudents extends Facilitator1819FormComponen
 
     return requiredFields;
   }
+
+  /**
+   * @override
+   */
+  static processPageData(data) {
```

Comment:
Maybe a future todo would be to improve the form controller to clear hidden fields as needed

Reply:
Yeah that would be nice. As it stands, we don't have a good way to discover which components are visible short of inspecting the DOM (or React component tree I suppose) for those names. However, we could improve the way we render conditional components in such a way that is more imperative, maybe with a container component or function, that will take care of the dynamic required fields and data processing. If I have some time after the initial Teacher Application implementation I'll take a look at streamlining this.